apis/bigquerydatapolicy/v1alpha1/datapolicy_identity.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common"
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DataPolicyIdentity defines the resource reference to BigQueryDataPolicy, which "External" field
+// holds the GCP identifier for the KRM object.
+type DataPolicyIdentity struct {
+	parent *DataPolicyParent
+	id     string
+}
+
+func (i *DataPolicyIdentity) String() string {
+	return i.parent.String() + "/datapolicies/" + i.id
```

Comment:
I think it should be capital `P`, `dataPolicies`. But we can fix this in later PRs.